core/templates/domain/editor/undo_redo/ChangeObjectFactory.ts

```diff
@@ -28,22 +28,57 @@ import { Injectable } from '@angular/core';
 
 import { MisconceptionBackendDict } from
   'domain/skill/MisconceptionObjectFactory';
+import { RecordedVoiceOverBackendDict } from
+  'domain/exploration/RecordedVoiceoversObjectFactory';
+import { StateBackendDict } from
+  'domain/state/StateObjectFactory';
+import { SubtitledHtmlBackendDict } from
+  'domain/exploration/SubtitledHtmlObjectFactory';
+import { WorkedExampleBackendDict } from
+  'domain/skill/WorkedExampleObjectFactory';
+
+interface CollectionTitleChange {
+  'cmd': 'edit_collection_property';
+  'property_name': 'title';
+  'new_value': string;
+  'old_value': string;
+}
 
-interface CollectionPropertyChange {
+interface CollectionCategoryChange {
```

Comment:
The changes here are much better -- and, as you've seen, they've caught some errors that would have gone undetected otherwise. Thanks @nishantwrp!